x/campaign/types/params_test.go

```diff
@@ -0,0 +1,87 @@
+package types
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParamsValidateBasic(t *testing.T) {
+	tests := []struct {
+		name   string
+		params Params
+		err    error
+	}{
+		{
+			name:   "invalid range",
+			params: NewParams(DefaultMaxTotalSupply, DefaultMinTotalSupply),
+			err:    errors.New("maximum total supply should be greater than greater or equal than minimum total supply"),
+		},
+		{
+			name:   "valid range",
+			params: NewParams(DefaultMinTotalSupply, DefaultMaxTotalSupply),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := tt.params.ValidateBasic()
+			if tt.err != nil {
+				require.Error(t, err, tt.err)
+				require.Equal(t, err, tt.err)
+				return
+			}
+			require.NoError(t, err)
+		})
+	}
+}
+
+func TestValidateTotalSupplyRange(t *testing.T) {
+	tests := []struct {
+		name        string
+		supplyRange interface{}
+		err         error
+	}{
+		{
+			name:        "invalid interface",
+			supplyRange: "test",
+			err:         fmt.Errorf("invalid parameter type: string"),
+		},
+		{
+			name: "invalid minTotalSupply",
+			supplyRange: TotalSupplyRange{
+				MinTotalSupply: sdk.ZeroInt(),
+				MaxTotalSupply: DefaultMaxTotalSupply,
+			},
+			err: errors.New("parameter minTotalSupply cannot be less than one"),
+		},
+		{
+			name: "invalid range",
+			supplyRange: TotalSupplyRange{
+				MinTotalSupply: DefaultMaxTotalSupply,
+				MaxTotalSupply: DefaultMinTotalSupply,
+			},
+			err: errors.New("parameter maxTotalSupply cannot be less than minTotalSupply"),
```

Comment:
They're actually equal in this case, and I don't think we should prevent this situation, even though it shouldn't be set like this in practice, this means there is a single total supply value that can be set of all coin

Reply:
They aren't equal (`DefaultMinTotalSupply` used for max and `DefaultMaxTotalSupply` used for min hence max < min (?) ), and the situation in which the two range boundaries are equal is allowed.

Reply:
Sorry I misread